src/assets/scripts/client/aircraft/AircraftModel.js

```diff
@@ -1457,18 +1545,48 @@ export default class AircraftModel {
             //     return;
 
             case MCP_MODE.ALTITUDE.VNAV: {
-                const waypointAltitude = this.fms.currentWaypoint.altitudeRestriction;
-                const waypointHasAltitude = waypointAltitude !== -1;
+                const hardRestrictedWaypointModel = this.fms.nextHardAltitudeRestrictedWaypoint;
```

Comment:
Moved to:
- `AircraftModel._calculateTargetedAltitudeVnav()`, which calls...
- `AircraftModel._calculateTargetedAltitudeVnavClimb()` and
- `AircraftModel._calculateTargetedAltitudeVnavDescent()`

done